### PR TITLE
Refactor LockExtensionTest to use Prophecy

### PR DESCRIPTION
**Pull request by jordisala1991**
### Subject

Refactored LockExtensionTest to use Prophecy.

### To do

- [x] My PR stuff
- [x] Update the tests

### CODE REVIEW

File: Tests/Admin/Extension/LockExtensionTest.php [hunk 1]
```diff
@@ -52,240 +51,179 @@ class LockExtensionTest extends \PHPUnit_Framework_TestCase
      */
     private $request;
 
-    public function setUp()
+    protected function setUp()
```
Comment (greg0ire) [inline Tests/Admin/Extension/LockExtensionTest.php]: Good change, but should be in a separate commit (use `git add --patch`)

File: Tests/Admin/Extension/LockExtensionTest.php [hunk 2]
```diff
@@ -52,240 +51,179 @@ class LockExtensionTest extends \PHPUnit_Framework_TestCase
...
+        $form = $this->prophesize('Symfony\Component\Form\FormInterface');
+
+        $form->getData()
+            ->willReturn($this->object);
 
-        $form->expects($this->any())
-            ->method('getData')
-            ->will($this->returnValue($this->object));
+        $form->getParent()
+            ->willReturn(null);
```
Comment (greg0ire) [inline Tests/Admin/Extension/LockExtensionTest.php]: How about writing this on one line? It will show even more how concise prophecy is.

**Comment by greg0ire** [comment]
Nice contribution @jordisala1991! @Soullivaneuh , please have a look, this may convince you to change to prophecy ;)

File: Tests/Admin/Extension/LockExtensionTest.php (hunk 2 above)
Comment (jordisala1991) [inline Tests/Admin/Extension/LockExtensionTest.php]: I though about doing in one line, but I wasn't sure. I think almost all the mocked methods can be written in one line. Should I rewrite all of them in one line?

File: Tests/Admin/Extension/LockExtensionTest.php (hunk 1 above)
Comment (jordisala1991) [inline Tests/Admin/Extension/LockExtensionTest.php]: Done

File: Tests/Admin/Extension/LockExtensionTest.php (hunk 2 above)
Comment (greg0ire) [inline Tests/Admin/Extension/LockExtensionTest.php]: I would say : only if they fit under 120 chars

**Comment by Soullivaneuh** [comment]
Yeah I'll take a look after my refactoring but please don't merge it before a the dev-kit issue is closed.

@jordisala1991 Could you BTW link this PR from the dev-kit issue? Thanks for your work. :+1:

File: Tests/Admin/Extension/LockExtensionTest.php [hunk 3]
```diff
@@ -52,240 +51,155 @@ class LockExtensionTest extends \PHPUnit_Framework_TestCase
...
-                $this->equalTo(array(
-                    'mapped' => false,
-                    'data' => 1,
-                ))
-            );
+        $this->configureAdmin(null, null, $this->modelManager->reveal());
+        $event = new FormEvent($form->reveal(), array());
 
-        $this->lockExtension->configureFormFields($formMapper);
+        $this->modelManager->getLockVersion(array())->willReturn(1);
```
Comment (Soullivaneuh) [inline Tests/Admin/Extension/LockExtensionTest.php]: This why I'm hesitating.

`$this->modelManager` is a mock of `ModelManager` class. So how the IDE can find auto-completion for both `getLockVersion` and `willReturn`? Before that, do the auto-completion works for both?

Comment (greg0ire) [inline Tests/Admin/Extension/LockExtensionTest.php]: Could someone maybe try https://github.com/maxfilatov/phpuaca , see if it does miracles ? Anyway, I don't really think auto-completion can really be a blocker, given that even without it, you will probably be quicker than with basic mocks and auto-completion.

Comment (Soullivaneuh) [inline Tests/Admin/Extension/LockExtensionTest.php]: I'll give a try.

But IDE completion **is** important IMHO. You will loose a lot of time if this does not work.

Comment (greg0ire) [inline Tests/Admin/Extension/LockExtensionTest.php]: > You will loose a lot of time if this does not work.

Well not me, I use vim :trollface: and local auto-completion makes gives me some for prophecy ^^

Comment (jordisala1991) [inline Tests/Admin/Extension/LockExtensionTest.php]: Tried right now. Is doesn't do miracles... The willReturn part is not detected, and the $this->prophesize is also not detected

Comment (Soullivaneuh) [inline Tests/Admin/Extension/LockExtensionTest.php]: Did you tried with the plugin linked by @greg0ire?

Comment (jordisala1991) [inline Tests/Admin/Extension/LockExtensionTest.php]: Yes.. I'm not an expert of PHPStorm but since the part where you do:

    $this->modelManager->getLockVersion()

getLockVersion is not a method defined in ObjectProphecy. It is a __call method. Does PHPStorm detect this kind of calls?

Also when you do the `->reveal()` It is returning an `object`, probably extending or implementing the class of your mock but PHPStorm doesn't detect it either.

**Comment by OskarStark** [comment]
@Soullivaneuh stuck StyleCi build

What is the next step for this PR? The dev-kit issue here: https://github.com/sonata-project/dev-kit/issues/89 has no decision... :(

**Comment by greg0ire** [comment]
> What is the next step for this PR? The dev-kit issue here: sonata-project/dev-kit#89 has no decision... :(

The next step is making a decision and documenting it in the CONTRIBUTING.md

**Comment by OskarStark** [comment]
> The next step is making a decision and documenting it in the CONTRIBUTING.md

I know, but when... 😢

**Comment by OskarStark** [comment]
@greg0ire can we merge this after the decision?

**Comment by greg0ire** [comment]
LGTM Please rebase so that StyleCI passes.

**Comment by greg0ire** [comment]
ping @jordisala1991

**Comment by jordisala1991** [comment]
I will rebase today, sorry!

**Comment by greg0ire** [comment]
> I will rebase today, sorry!

Don't be, there's really no hurry ;)

**Comment by greg0ire** [comment]
> Don't be, there's really no hurry ;)

But on the other hand, if you *could* rebase, it would be… great!

**Comment by greg0ire** [comment]
ping @jordisala1991

**Comment by core23** [comment]
Can you have a look at StyleCi @Soullivaneuh

**Comment by greg0ire** [comment]
> Can you have a look at StyleCi @Soullivaneuh

No need to bother @Soullivaneuh , this should be fixed by a rebase.

**Comment by jordisala1991** [comment]
Done, sorry for the delay!

**Comment by greg0ire** [comment]
Aaand it's stuck, just like #4134

**Comment by jordisala1991** [comment]
What do you mean? Bug on github?

**Comment by greg0ire** [comment]
Yes, see #4134

**Comment by greg0ire** [comment]
Ok we can merge! Dunno if it is thanks to @Soullivaneuh or if github solved their bug.

**Comment by greg0ire** [comment]
Thanks @jordisala1991 !
